test/utils/wasm3_engine.cpp

```diff
@@ -38,7 +39,8 @@ class Wasm3Engine final : public WasmEngine
 
 namespace
 {
-const void* env_adler32(IM3Runtime /*runtime*/, uint64_t* stack, void* mem)
+const void* env_adler32(
```

Comment:
Based on https://github.com/wasm3/wasm3/blob/main/source/wasm3.h#L327-L340 this would be the correct way:
```c
m3ApiRawFunction(env_adler32)
{
    (void)runtime;
    (void)_ctx;

    m3ApiReturnType (uint32_t)
    m3ApiGetArgMem  (const uint8_t*, offset)
    m3ApiGetArg     (uint32_t, length)

    // This would be the correct way, but it traps. Although the numbers seem to be correct.
    // mem + offset + length <= mem + m3_getMemorySize
    m3ApiCheckMem(offset, length);

    m3ApiReturn(fizzy::test::adler32({offset, length}));
}
```

See also an example wasi function: https://github.com/wasm3/wasm3/blob/main/source/m3_api_meta_wasi.c#L146